src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java

```diff
@@ -425,4 +439,26 @@ private void sendUpdateSingleCaseECC(String authToken, CaseDetails currentCaseDe
         }
     }
 
+    /**
+     * Calls reference data API to add HMCTSServiceId to supplementary_data to a case.
+     * @param caseDetails Details on the case
+     * @param accessToken authorisation token for reference data api
+     */
+    public void setHmctsServiceIdSupplementary(CaseDetails caseDetails, String accessToken) throws IOException {
+        Map<String, Map<String, Object>> payloadData = Maps.newHashMap();
+        payloadData.put("$set", singletonMap(HMCTS_SERVICE_ID, hmctsServiceId));
+
+        Map<String, Object> payload = Maps.newHashMap();
+        payload.put("supplementary_data_updates", payloadData);
+
+        try {
+            var response = ccdClient.setSupplementaryData(accessToken, payload, caseDetails.getCaseId());
+            if (response == null) {
+                throw new CaseCreationException("Call to Supplementary Data API did not return successfully");
+            }
+            log.info("Http status received from CCD supplementary update API; {}", response.getStatusCodeValue());
+        } catch (RestClientResponseException e) {
+            throw new CaseCreationException("Call to Supplementary Data API failed with " + e.getMessage());
```

Comment:
Do we want to log an error/warning if it fails to add service id to supplementary data rather than just throwing an exception?

Reply:
Exceptions are logged anyway, wouldn't adding a log here just be redundant?